src/components/CartList/CartList.jsx

```diff
@@ -1,27 +1,135 @@
   const onMoveFromCartToWIshlist = (product) => {
-    wishlistDispatch({ type: "ADD", payload: product });
-    cartDispatch({ type: "DELETE", payload: product });
+    moveRequest({token: authState.token, product, action: "CART_TO_WISHLIST"});
   };
+  const onRemoveFromCart = (product) => {
+    sendRequest({ product, token: authState.token });
+  };
+  const changeItemQuantity = (product, type) => {
+    changeRequest({ product, token: authState.token, type });
+  };
+  const isInWishlist = (product) => {
+    return wishlist.find((item) => item._id === product._id);
+  };
+  useEffect(() => {
+    if (status === "completed") {
+      cartDispatch({ type: "REMOVE", payload: data });
+      toast.success("Item removed from cart!", {
+        position: "top-right",
+        autoClose: 1000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+    if (status === "error") {
+      toast.error("Oops! There was some error!", {
+        position: "top-right",
+        autoClose: 1000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+  }, [status, cartDispatch, data]);
+  useEffect(() => {
+    if (changeStatus === "completed") {
+      cartDispatch({ type: "CHANGE_QUANTITY", payload: changeData });
+      toast.success("Quantity updated", {
+        position: "top-right",
+        autoClose: 500,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+    if (changeStatus === "error") {
+      toast.error("Oops! Try again!", {
+        position: "top-right",
+        autoClose: 500,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+  }, [changeStatus, changeData, cartDispatch]);
+  useEffect(()=>{
+    if(moveStatus==="completed"){
+      wishlistDispatch({ type: "ADD", payload: moveData.wishlist });
+      cartDispatch({ type: "REMOVE", payload: moveData.cart });
+      toast.success("Item moved from cart to wishlist!", {
+        position: "top-right",
+        autoClose: 500,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+    if(moveData==="error"){
+      toast.error("Oops! Try again!", {
+        position: "top-right",
+        autoClose: 500,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: false,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+  },[wishlistDispatch, cartDispatch, moveStatus, moveData])
   return (
     <>
       <div className={styles["cart-products"]}>
         {cart.cart.map((item) => (
```

Comment:
Suggestion: You could use some different name instead of `cart.cart` as it confuses the reader.